x-pack/plugins/monitoring/server/routes/index.ts

```diff
@@ -4,14 +4,23 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
 /* eslint import/namespace: ['error', { allowComputed: true }]*/
+
+import { MonitoringConfig } from '../config';
+import { decorateDebugServer } from '../debugLogger';
+import { RouteDependencies } from '../types';
 // @ts-ignore
 import * as uiRoutes from './api/v1/ui'; // namespace import
-import { RouteDependencies } from '../types';
 
-export function requireUIRoutes(server: any, npRoute: RouteDependencies) {
+export function requireUIRoutes(
+  _server: any,
+  config: MonitoringConfig,
+  npRoute: RouteDependencies
+) {
   const routes = Object.keys(uiRoutes);
+  const server = config.ui.debug_mode
```

Comment:
Check 1 of 2 to make sure we only decorate this server object if debug_mode is on...